### PR TITLE
[got] Update to 8.3.0

### LINKED ISSUE #23454: [@types/got] Need Update
- [x] I tried using the `@types/got` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @LinusU @BendingBender 

----------------------

`got` had a large break changes in `v8.x.x` API.
This Return: 
```
Promise<Response>
```
Instead of Return
```
Promise<Body>
```
I Hope you can update it soon.

### PR DESCRIPTION
**Pull request by ikokostya**
Fixes #23454

**Notable changes**

* Add support for WHATWG URL to `got` function.
* Add `got.CancelError` error type.
* Add `downloadProgress`, `uploadProgress` events.
* Add `cache` option to `GotOptions` and `fromCache` property to `Response` object.
* Add `throwHttpErrors` option to `GotOptions`.
* Add support for agent object to `GotOptions`.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/sindresorhus/got/releases
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/got/index.d.ts [hunk 1]
```diff
@@ -98,18 +104,23 @@ declare namespace got {
         json?: boolean;
     }
 
-    interface TimoutRequestOptions extends http.RequestOptions {
+    interface RequestOptions extends http.RequestOptions {
```
Comment (ikokostya) [inline types/got/index.d.ts]: This is potentially breaking change, but I think it's not a problem for major release of definition.

File: types/got/index.d.ts [hunk 2]
```diff
@@ -59,13 +64,14 @@ declare class StdError extends Error {
...
     Record<'get' | 'post' | 'put' | 'patch' | 'head' | 'delete', got.GotFn> &
     {
-        stream: got.GotStreamFn & Record<'get' | 'post' | 'put' | 'patch' | 'head' | 'delete', got.GotStreamFn>
-        RequestError: typeof RequestError
-        ReadError: typeof ReadError
-        ParseError: typeof ParseError
-        HTTPError: typeof HTTPError
-        MaxRedirectsError: typeof MaxRedirectsError
-        UnsupportedProtocolError: typeof UnsupportedProtocolError
+        stream: got.GotStreamFn & Record<'get' | 'post' | 'put' | 'patch' | 'head' | 'delete', got.GotStreamFn>;
```
Comment (ikokostya) [inline types/got/index.d.ts]: There I just add semicolon `;` to the end of each line. If this change is redundant I can revert it.

File: types/got/index.d.ts (hunk 1 above)
Comment (LinusU) [inline types/got/index.d.ts]: Maybe we should name this something "strange" to discourage people from using it? It's only here to work around how we are extending the options...

`_InternalRequestOptions`?

**Comment by LinusU** [comment]
One small thing, otherwise very good!

File: types/got/index.d.ts (hunk 1 above)
Comment (ikokostya) [reply types/got/index.d.ts]: Yes, I also think about this. We move this interface from namespace.

Comment (LinusU) [reply types/got/index.d.ts]: Perfect 👍

**Review by LinusU — APPROVED** [review]
Great 👍

File: types/got/index.d.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for got 7.1
+// Type definitions for got 8.3
 // Project: https://github.com/sindresorhus/got#readme
 // Definitions by: BendingBender <https://github.com/BendingBender>
 //                 Linus Unnebäck <https://github.com/LinusU>
```
Comment (LinusU) [inline types/got/index.d.ts]: If you want to, you can add yourself to this list. That way you'll be able to help review pull requests to this package in the future ☺️

**Comment by LinusU** [comment]
Thanks for the great work!

**Pull request merged**